web/src/test/java/org/cbioportal/genome_nexus/web/CancerHotspotsIntegrationTest.java

```diff
@@ -0,0 +1,191 @@
+package org.cbioportal.genome_nexus.web;
+
+import org.cbioportal.genome_nexus.model.AggregatedHotspots;
+import org.cbioportal.genome_nexus.model.GenomicLocation;
+import org.cbioportal.genome_nexus.model.Hotspot;
+import org.cbioportal.genome_nexus.service.annotation.NotationConverter;
+import org.junit.Test;
+import org.junit.runner.RunWith;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.boot.test.context.SpringBootTest;
+import org.springframework.test.context.junit4.SpringJUnit4ClassRunner;
+import org.springframework.web.client.RestTemplate;
+
+import java.util.List;
+import java.util.Arrays;
+import java.util.stream.Collectors;
+
+import static org.junit.Assert.assertEquals;
+
+@RunWith(SpringJUnit4ClassRunner.class)
+@SpringBootTest(
+    webEnvironment = SpringBootTest.WebEnvironment.DEFINED_PORT,
+    properties = {
+        "vep.url=http://grch37.rest.ensembl.org/vep/human/hgvs/VARIANT?content-type=application/json&xref_refseq=1&ccds=1&canonical=1&domains=1&hgvs=1&numbers=1&protein=1",
+        "hotspots.url=http://cancerhotspots.org/api/hotspots/single/",
+        "hotspots.3d.url=http://cancerhotspots.org/3d/api/hotspots/3d/",
+        "spring.data.mongodb.uri=mongodb://localhost/integration",
+        "server.port=38888"
+    }
+)
+public class CancerHotspotsIntegrationTest
+{
+    private RestTemplate restTemplate = new RestTemplate();
+
+    @Autowired
+    NotationConverter notationConverter;
+
+    @Test
+    public void testSingleResidueHotspots()
+    {
+        String baseUrl = "http://localhost:38888/cancer_hotspots/genomic/";
+
+        // genomic location strings (needed for GET requests)
+        String[] genomicLocations = {
+            "7,140453193,140453193,T,C", // recurrent and 3D
+            "17,37880220,37880220,T,C", // recurrent
+            "17,37879794,37879794,G,C", // recurrent
+            "12,25398284,25398285,CC,GA", // recurrent and 3D
+            "12,25380271,25380271,C,T", // 3D
+            "10,89692940,89692940,C,T" // none
```

Comment:
@onursumer should we handle `chr1` as well?

Reply:
things prefixed with `chr*` basically

Reply:
good point, probably we are not handling the `chr*` case right now.

Reply:
yeah not a big deal, but would be good to include. Also for indels the format is different for vcf vs maf. So good to have some parameter to indicate which one we are using